### ISSUE TITLE
Remove Binary Search

### ISSUE THREAD

**Issue by jjatie** [danielgindi/Charts#3640]
* [ x] I've read, understood, and done my best to follow the [*CONTRIBUTING guidelines](https://github.com/jjatie/Charts/blob/master/CONTRIBUTING.md).

## Background
Binary search provides optimized search for sorted collections. For unsorted collections, binary search is significantly slower than iterating over the collection. Charts implements binary search, however it does not enforce sorted collections, causing significantly slower lookup times for unsorted collections.

## Proposal
Remove binary search until such a time that we can a) enforce sorted chart data or b) have sorted and unsorted chart data with different look up algorithms.

## Charts Environment

**Charts version: 3.2
**Xcode version:10
**Swift version:4.2

**Comment by liuxuan30** [comment]
I remember it's sorted back in time. You sure we are binary searching unsorted collections? Did we change it to unsorted by mistake?

**Comment by jjatie** [comment]
@liuxuan30 Yes, we perform binary search on all sets with no guarantee they are sorted.

**Comment by liuxuan30** [comment]
If so, we indeed need to fix. But I guess the search will have high false rate? I wonder why no people complain about this. especially when highlighting.
